react/src/components/UserSettingModal.tsx

```diff
@@ -172,49 +176,136 @@ const UserSettingModal: React.FC<Props> = ({
+        $isNotSupportTotp: Boolean!
+      ) {
+        create_user(email: $email, props: $props) {
+          ok
+          msg
+          user {
+            id
+            email
+            username
+            need_password_change
+            full_name
+            description
+            status
+            domain_name
+            role
+            groups {
+              id
+              name
+            }
+            resource_policy
+            sudo_session_enabled
+              @skipOnClient(if: $isNotSupportSudoSessionEnabled)
+            totp_activated @skipOnClient(if: $isNotSupportTotp)
+            ...TOTPActivateModalFragment
+          }
+        }
+      }
+    `);
+
   const mutationToRemoveTotp = useTanMutation({
     mutationFn: (email: string) => {
       return baiClient.remove_totp(email);
     },
   });
 
-  const [isOpenTOTPActivateModal, { toggle: toggleTOTPActivateModal }] =
-    useToggle(false);
-
-  const _onOk = () => {
-    formRef.current?.validateFields().then(async (values) => {
-      let input = { ...values };
-      delete input.email;
-      input = _.omit(input, ['password_confirm']);
-      input = _.omitBy(input, (item) => item === undefined || item === '');
-      if (!sudoSessionEnabledSupported) {
-        delete input?.sudo_session_enabled;
-      }
-      // TOTP setting
-      if (!isTOTPSupported) {
-        delete input?.totp_activated;
-      }
+  const INITIAL_VALUES = {
+    email: '',
+    username: '',
+    need_password_change: false,
+    full_name: '',
+    description: '',
+    status: 'active',
+    domain_name: currentDomainName,
+    role: 'user',
+    is_active: true,
+    resource_policy: 'default',
+  };
 
-      commitModifyUserSetting({
-        variables: {
-          email: values?.email || '',
-          props: input,
-          isNotSupportSudoSessionEnabled: !sudoSessionEnabledSupported,
-          isNotSupportTotp: !isTOTPSupported,
-        },
-        onCompleted(res) {
-          if (res?.modify_user?.ok) {
-            message.success(t('environment.SuccessfullyModified'));
-          } else {
-            message.error(res?.modify_user?.msg);
-          }
-          dispatchEvent('ok', null);
-        },
-        onError(err) {
-          message.error(err?.message);
-        },
-      });
-    });
+  const handleOk = () => {
+    formRef.current
+      ?.validateFields()
+      .then(async (values) => {
+        if (user) {
+          const props: ModifyUserInput = _.omitBy(
+            _.omit(values, ['email', 'password_confirm']),
+            (item) => _.includes([undefined, ''], item),
+          );
+          commitModifyUserSetting({
+            variables: {
+              email: values?.email || '',
+              props: props,
+              isNotSupportSudoSessionEnabled: !sudoSessionEnabledSupported,
+              isNotSupportTotp: !isTOTPSupported,
+            },
+            onCompleted: (res) => {
```

Comment:
```suggestion
            onCompleted: (res, errors) => {
```
Let's do more error handling.